src/test/java/ru/testtask/service/FileServiceTest.java

```diff
@@ -0,0 +1,163 @@
+        UploadFileDTO uploadFileDTO = UploadFileDTO.builder()
+                .fileUrl("https://www.zastavki.com/pictures/2560x1600/2011/Space_Big_planet_031405_.jpg")
+                .fileName("b").build();
+        fileService.uploadFile(uploadFileDTO);
+    }
+
+    @Test
+    public void moveFileTest() throws IOException {
+        FileData fileData = testUtils.createTestFileDataWithFile();
+        Path sourcePath = fileService.getFileAbsolutePath(fileData);
+        fileData = fileService.moveFile(fileData.getId(), "a/b/c");
+        assertFalse(Files.exists(sourcePath));
+        assertTrue(Files.exists(fileService.getStorageRootPath()
+                .resolve("a/b/c").resolve(fileData.getFilename())));
+        assertEquals("a/b/c", fileData.getDirectory());
+    }
+
+    @Test
+    public void copyFileTest() throws IOException {
+        FileData fileData = testUtils.createTestFileDataWithFile();
+        FileData copiedFileData = fileService.copyFile(fileData.getId(), "dir1");
+        copiedFileData = fileService.getFileById(copiedFileData.getId());
+        assertEquals("dir1", copiedFileData.getDirectory());
+        assertTrue(Files.exists(fileService.getFileAbsolutePath(copiedFileData)));
+        assertEquals(2, fileService.searchByRegex("test").size());
+        fileService.copyFile(fileData.getId(), "dir1");
+        assertEquals(3, fileService.searchByRegex("test").size());
+        assertNotNull(fileService.searchByRegex("testFile (0)."));
+    }
+
+    @Test
+    public void findByRegexpTest(){
+        testUtils.createTestDataForRegexTest();
+
+        List<FileData> fileDataList = fileService.searchByRegex("^abc");
+        assertEquals(1, fileDataList.size());
+        assertEquals("abcdefg", fileDataList.get(0).getOriginalFilename());
+
+        fileDataList = fileService.searchByRegex("abc");
+        assertEquals(2, fileDataList.size());
+
+        fileDataList = fileService.searchByRegex("\\d");
+        assertEquals(1, fileDataList.size());
+        assertEquals("12345", fileDataList.get(0).getOriginalFilename());
+
+        fileDataList = fileService.searchByRegex(".");
+        assertEquals(3, fileDataList.size());
+    }
+
+
+    @Test
+    public void findByDirectoryTest(){
+        testUtils.createTestDataForRegexTest();
+
+        List<FileData> fileDataList = fileService.getFileListFromDirectory("dir1");
+        assertEquals(1, fileDataList.size());
+        assertEquals("abcdefg", fileDataList.get(0).getOriginalFilename());
+
+        fileDataList = fileService.getFileListFromDirectory("dir1/dir2");
+        assertEquals(1, fileDataList.size());
+
+        fileDataList = fileService.getFileListFromDirectory("dir3");
+        assertEquals(1, fileDataList.size());
+
+        fileDataList = fileService.getFileListFromDirectory("folder");
+        assertEquals(1, fileDataList.size());
+    }
+
+    @Test
+    public void multithreadingCopyTest() throws IOException, InterruptedException {
```

Comment:
We have incorrect condition of the db for this test. Unique index will not be created